### PR TITLE
change todoattachment union into original file attachment

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

**Review by msyyc — APPROVED** [review]

File: todoApp/spec/main.tsp
Comment (weidongxu-microsoft) [inline todoApp/spec/main.tsp:1]: Can we do this?

```ts
@jsonSchema
model TodoFileBytesAttachment extends TodoAttachment {
  kind: TodoAttachmentType.File;
  /** The file contents */
  file: bytes;
}

@jsonSchema
model TodoUrlAttachment extends TodoAttachment {
  kind: TodoAttachmentType.Url;
  /** A description of the URL */
  description: string;

  /** The url */
  url: url;
}

@jsonSchema
model TodoAttachment {
  /** discriminator kind */
  kind: TodoAttachmentType;
}

@jsonSchema
union TodoAttachmentType {
  string,
  File: "file",
  Url: "url",
}
```

File: todoApp/spec/main.tsp [hunk 1]
```diff
@@ -104,20 +104,37 @@ model TodoLabelRecord {
...
+model TodoUrlAttachment extends TodoAttachment {
+  kind: "url";
   /** A description of the URL */
   description: string;
 
   /** The url */
   url: url;
 }
 
+
```
Comment (qiaozha) [inline todoApp/spec/main.tsp]:
```suggestion
@jsonSchema
```

File: todoApp/spec/main.tsp [hunk 2]
```diff
@@ -104,20 +104,36 @@ model TodoLabelRecord {
   color?: string;
 }
 
-union TodoAttachment {
-  file: bytes,
-  url: TodoUrlAttachment,
+@jsonSchema
+model TodoFileBytesAttachment extends TodoAttachment {
+  kind: "file";
```
Comment (qiaozha) [inline todoApp/spec/main.tsp]:
```suggestion
  kind: TodoAttachmentType.File;
```

File: todoApp/spec/main.tsp [hunk 3]
```diff
@@ -104,20 +104,36 @@ model TodoLabelRecord {
...
+  /** The file contents */
+  file: bytes;
 }
 
 @jsonSchema
-model TodoUrlAttachment {
+model TodoUrlAttachment extends TodoAttachment {
+  kind: "url";
```
Comment (qiaozha) [inline todoApp/spec/main.tsp]:
```suggestion
  kind: TodoAttachmentType.Url;
```

File: todoApp/spec/tspconfig.yaml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 emit:
  - "@typespec/http-server-csharp"
- - "@typespec/http-server-javascript"
- - "@typespec/http-client-csharp"
- - "@typespec/http-client-java"
+#  - "@typespec/http-server-javascript"
+#  - "@typespec/http-client-csharp"
+#  - "@typespec/http-client-java"
```
Comment (qiaozha) [inline todoApp/spec/tspconfig.yaml]:
```suggestion
  - "@typespec/http-server-javascript"
  - "@typespec/http-client-csharp"
  - "@typespec/http-client-java"
```

File: todoApp/spec/tspconfig.yaml [hunk 5]
```diff
@@ -1,8 +1,8 @@
 emit:
  - "@typespec/http-server-csharp"
- - "@typespec/http-server-javascript"
- - "@typespec/http-client-csharp"
- - "@typespec/http-client-java"
+  - "@typespec/http-server-javascript"
+  - "@typespec/http-client-csharp"
+  - "@typespec/http-client-java"
```
Comment (qiaozha) [inline todoApp/spec/tspconfig.yaml]:
```suggestion
 - "@typespec/http-server-javascript"
 - "@typespec/http-client-csharp"
 - "@typespec/http-client-java"
```

File: todoApp/spec/main.tsp [hunk 6]
```diff
@@ -118,6 +113,19 @@ model TodoUrlAttachment {
...
+
+  /** The media type of the attachment */
+  mediaType: string;
+
+  /** The contents of the file */
+  contents: bytes;
+}
+
 @jsonSchema
 model TodoFileAttachment is File;
```
Comment (weidongxu-microsoft) [inline todoApp/spec/main.tsp:130]: Mark seems saying in meeting that we no longer need `TodoFileAttachment`, and can directly put `HttpPart<File>` in `ToDoItemMultipartRequest` and `FileAttachmentMultipartRequest`. Did anyone remember same, or it is just me?

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**